test/socket_test.dart

```diff
@@ -1,26 +1,176 @@
+import 'dart:io';
+
 import 'package:test/test.dart';
 import 'package:realtime/realtime.dart';
-
 import 'package:web_socket_channel/io.dart';
 
+typedef IOWebSocketChannelClosure = IOWebSocketChannel Function(
+    String url, Map<String, String> headers);
+
 void main() {
-  test('sets defaults', () async {
-    final socket = Socket('wss://example.com/socket', logger: (kind, msg, data) => print('[$kind] $msg $data'));
-    expect(socket.channels.length, 0);
-    expect(socket.sendBuffer.length, 0);
-    expect(socket.ref, 0);
-    expect(socket.endPoint, 'wss://example.com/socket/websocket');
-    expect(socket.stateChangeCallbacks, {
-      'open': [],
-      'close': [],
-      'error': [],
-      'message': [],
-    });
-    expect(socket.transport, IOWebSocketChannel);
-    expect(socket.timeout, const Duration(milliseconds: 10000));
-    expect(socket.longpollerTimeout, 20000);
-    expect(socket.heartbeatIntervalMs, 30000);
-    expect(socket.logger is Logger, true);
-    expect(socket.reconnectAfterMs is Function, true);
+  HttpServer mockServer;
+
+  setUp(() async {
+    mockServer = await HttpServer.bind('localhost', 0);
+    mockServer.transform(WebSocketTransformer());
+  });
+
+  tearDown(() async {
+    if (mockServer != null) {
+      await mockServer.close();
+    } else {
+      print('mock server was null');
+    }
+  });
+
+  group('constructor', () {
+    test('sets defaults', () async {
+      final socket = Socket('wss://example.com/socket');
+      expect(socket.channels.length, 0);
+      expect(socket.sendBuffer.length, 0);
+      expect(socket.ref, 0);
+      expect(socket.endPoint, 'wss://example.com/socket/websocket');
+      expect(socket.stateChangeCallbacks, {
+        'open': [],
+        'close': [],
+        'error': [],
+        'message': [],
+      });
+      expect(socket.transport is IOWebSocketChannelClosure, true);
+      expect(socket.timeout, const Duration(milliseconds: 10000));
+      expect(socket.longpollerTimeout, 20000);
+      expect(socket.heartbeatIntervalMs, 30000);
+      expect(socket.logger is Logger, true);
```

Comment:
socket.logger is null as default

Reply:
I was thinking that rather than having a null value flying around, logger can be initialized with a function having empty body.
Something like

```
...
logger = (logger ?? (a, b, c) {}),
...
```